functions/Restore-DbaBackup.ps1

```diff
@@ -0,0 +1,215 @@
+function Restore-DbaBackup
```

Comment:
Hi @Stuart-Moore 

You may be working on one of our showpieces here.
So I'm rather strict and thorough, OK?

I do agree with the idea of creating modular functions by using more private functions, but I just experienced that makes reviewing a lot harder.
Maybe a flow chart and a lot of inline comments could help. 😄 

I haven't actually executed any restore yet, I'm just reading the code for now.
So some of my remarks are actually questions and assumptions.